sqlglot/optimizer/unnest_subqueries.py

```diff
@@ -89,35 +89,53 @@ def unnest(select, parent_select, sequence):
 
 
 def decorrelate(select, parent_select, external_columns, sequence):
+    if any(
```

Comment:
Sounds good, yeah. I think the reason I kept it in the top was so that I would propagate less arguments to each `decorrelate_*` helper, though I agree that this will reduce clutter.

Reply:
I changed `decorrelate` so that it does filters by default, and only branches to `decorrelate_projection` if the `any` condition is met (which was moved further down, as suggested).